posthog/hogql/database.py

```diff
@@ -328,4 +331,12 @@ def get_table(self, table_name: str) -> Table:
         raise ValueError(f'Table "{table_name}" not found in database')
 
 
-database = Database()
+def create_hogql_database(team_id: Optional[int]) -> Database:
```

Comment:
This was my initial thought as well, though I couldn't find a way to make that work _smoothly_. Granted, I didn't try much.

The problem is that it's easy to add this on `Database()`, but then I'd need to re-initialise all tables with `team_id`, and in all tables potentially initialize all fields with `team_id`, etc. I'd have to stop using defaults for the fields, and move all property intializations into a cascading set of __init__-s. This might be inevitable going forward, but I felt this small function is Good Enough ™ for now...

Reply:
I'm not really sure how `__new__` makes things nicer in this case. I can move that one `if PoE enabled` up into the `__init__` of `Database`, but how do I get it into all the tables, and further into all the fields?

![image](https://user-images.githubusercontent.com/53387/225710031-0499fc54-6732-4c1b-a1d2-d9824e6e49b7.png)

It'll have to be something like:

![image](https://user-images.githubusercontent.com/53387/225710679-b2facb08-1d7a-4a54-a981-f18b16f4c592.png)

or some other bunch of duplication.

Any ideas? 🤔 

I'd personally just defer it all since we're going to have to redo the architecture once we launch calculated properties and virtual tables, however we will call them.

Reply:
This doesn't really work since now I either need to 1) pass a default `events`, `persons`, etc when doing `Database()`... or set them as `Optional` and handle the `None`s everywhere. I couldn't find a way to make it better with `__new__` either. Feeling diminishing returns at this point.